daemon/api_users.go

```diff
@@ -321,7 +321,9 @@ func createUser(c *Command, createData postUserCreateData) Response {
 		return InternalError(err.Error())
 	}
 
-	if createData.singleUserResultCompat {
+	// We only handle singleUserResultCompat if the request
+	// was *not* to create all known system users.
+	if (!createData.Known || createData.Email != "") && createData.singleUserResultCompat {
```

Comment:
I'm confused, it's strange to do the check here, shouldn't it be done at the place that set the flag instead??

Reply:
Good point! Definitely makes more sense